core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -92,4 +102,42 @@ public static Map<String, String> propertiesWithPrefix(
         .filter(e -> e.getKey().startsWith(prefix))
         .collect(Collectors.toMap(e -> e.getKey().replaceFirst(prefix, ""), Map.Entry::getValue));
   }
+
+  public static Map<String, String> applySchemaChanges(
+      Map<String, String> properties,
+      List<String> deletedColumns,
+      Map<String, String> renamedColumns) {
+    if (!properties.keySet().stream()
+        .anyMatch(key -> COLUMN_PREFIX_PROPERTIES.stream().anyMatch(key::startsWith))) {
+      return properties;
+    } else {
+      Map<String, String> updatedProperties = Maps.newHashMap();
+      properties
+          .keySet()
+          .forEach(
+              key -> {
+                String prefix =
+                    COLUMN_PREFIX_PROPERTIES.stream()
+                        .filter(key::startsWith)
+                        .findFirst()
+                        .orElse(null);
+
+                if (prefix != null) {
+                  String columnAlias = key.replaceFirst(prefix, "");
+                  if (renamedColumns.get(columnAlias) != null) {
+                    // The name has changed.
+                    String newKey = prefix + renamedColumns.get(columnAlias);
+                    updatedProperties.put(newKey, properties.get(key));
+                  } else if (!deletedColumns.contains(columnAlias)) {
+                    // Copy over the original.
+                    updatedProperties.put(key, properties.get(key));
+                  }
```

Comment:
Nit, the old comment 'implicit drop' may still make sense to keep?

Reply:
Good catch!